lib/auth/default_current_user_provider.rb

```diff
@@ -222,9 +245,16 @@ def log_on_user(user, session, cookies, opts = {})
     @env[CURRENT_USER_KEY] = user
   end
 
-  def cookie_hash(unhashed_auth_token)
+  def cookie_hash(unhashed_auth_token, user)
+    cookie = DiscourseAuthCookie.new(
+      token: unhashed_auth_token,
+      user_id: user.id,
+      trust_level: user.trust_level,
+      timestamp: Time.zone.now,
+      valid_for: SiteSetting.maximum_session_age.hours
```

Comment:
Just curious, do we need to store `timestamp` and `valid_for` separately? Can we replace it with a `valid_until_timestamp` field?